crates/ruff_python_formatter/resources/test/fixtures/ruff/statement/match.py

```diff
@@ -236,3 +236,21 @@ def foo():
         "b",
     ]:
         pass
+
+match foo:
+    case [1, 2, *rest]:
+        pass
+    case [1, 2, *_]:
+        pass
+    case [
+        1,
+        2,
+        *rest,
+    ]:
+        pass
+    case [1, 2, * # comment
+        rest]:
+        pass
+    case [1, 2, * # comment
+        _]:
+        pass
```

Comment:
This is formatted to:

```
    case [
        "NOT_YET_IMPLEMENTED_PatternMatchValue",
        "NOT_YET_IMPLEMENTED_PatternMatchValue",
        *rest  # comment
        ,
    ]:
        pass
    case [
        "NOT_YET_IMPLEMENTED_PatternMatchValue",
        "NOT_YET_IMPLEMENTED_PatternMatchValue",
        *_  # comment
        ,
    ]:
        pass
```

then formatted to:

```
    case [
        "NOT_YET_IMPLEMENTED_PatternMatchValue",
        "NOT_YET_IMPLEMENTED_PatternMatchValue",
        *rest,  # comment
    ]:
        pass
    case [
        "NOT_YET_IMPLEMENTED_PatternMatchValue",
        "NOT_YET_IMPLEMENTED_PatternMatchValue",
        *_,  # comment
    ]:
        pass
```

Reply:
If you look at
```python
match foo:
    case [* # comment
        rest, 1]:
        pass
    case [*rest # comment
          , 1]:
        pass
```
you get
```
{
    Node {
        kind: PatternMatchStar,
        range: 21..45,
        source: `* # comment⏎`,
    }: {
        "leading": [],
        "dangling": [
            SourceComment {
                text: "# comment",
                position: EndOfLine,
                formatted: true,
            },
        ],
        "trailing": [],
    },
    Node {
        kind: PatternMatchStar,
        range: 74..79,
        source: `*rest`,
    }: {
        "leading": [],
        "dangling": [],
        "trailing": [
            SourceComment {
                text: "# comment",
                position: EndOfLine,
                formatted: true,
            },
        ],
    },
}
```
so i think the best solution is to manually reassign the comment from dangling to trailing.

Reply:
@harupy - I typically run something like `cargo run foo.py --emit=stdout --print-comments` from `ruff/crates/ruff_python_formatter`.

Reply:
Oh sorry. You're asking how to modify these assignments. Any such modifications need to be done in `placement.rs` -- those are essentially custom rules that let us change the association of comments. See, e.g., `handle_trailing_expression_starred_star_end_of_line_comment` which is somewhat similar.

Reply:
You have to add a case in `handle_enclosed_comment` for when the enclosing node is `PatternMatchStar`. The general logic to fix comment placement is in `placement.rs`/`place_comment`.